src/components/movie-title/movie-title.test.tsx

```diff
@@ -0,0 +1,46 @@
+import '@testing-library/jest-dom'
+import { render, screen } from '@testing-library/react'
+import { MovieTitle } from './movie-title.tsx'
+import { describe, expect, it, vi } from 'vitest'
+import userEvent from '@testing-library/user-event'
+
+const mockProps = {
+  imgURL: 'https://example.com/image.jpg',
+  name: 'Inception',
+  year: 2010,
+  genres: ['Action', 'Sci-Fi'],
+  onClick: vi.fn(),
+}
+
+const user = userEvent.setup()
+
+describe('MovieTitle Component', () => {
+  it('renders correctly with given props', () => {
+    render(<MovieTitle {...mockProps} />)
+
+    expect(screen.getByText('Inception')).toBeInTheDocument()
+    expect(screen.getByText('2010')).toBeInTheDocument()
+    expect(screen.getByText('Action, Sci-Fi')).toBeInTheDocument()
+    expect(screen.getByRole('img', { name: /Inception/i })).toHaveAttribute('src', mockProps.imgURL)
+  })
+
+  it('calls onClick when container is clicked', async () => {
+    render(<MovieTitle {...mockProps} />)
+    await user.click(screen.getByText('Inception'))
+    expect(mockProps.onClick).toHaveBeenCalled()
```

Comment:
If you are using a mock function like this, you must clear mocks after each test.
And it's better to check how many times the function has been called for additional safeguarding against uncleaned mocks:

```suggestion
    expect(mockProps.onClick).toHaveBeenCalledTimes(1)
```